Consolidate browsers/permissions/accounts/wallet db in a single database

dshulyak opened the pull request:
closes: https://github.com/status-im/status-go/issues/1536

dshulyak committed:
Create single database appdatase

ghost commented:
<!--prchecklist--> 
### Pull Request Checklist
- [ ] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://docs.status.im/docs/build_status_go.html))?

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (33)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :x: | 15cfd711 | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1540/1/) | 2019-07-25 05:36:48 | ~36 sec | `android` | [:page_facing_up: log](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1540/1/consoleText) |
| :x: | 15cfd711 | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1540/1/) | 2019-07-25 05:36:48 | ~35 sec | `ios` | [:page_facing_up: log](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1540/1/consoleText) |
| :heavy_check_mark: | 15cfd711 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1540/1/) | 2019-07-25 05:37:03 | ~50 sec | `linux` | [:package: zip](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-190725-053613-15cfd7.zip) |
| | | | | | | |
| :heavy_check_mark: | 20bd0ddc | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1540/2/) | 2019-07-26 05:22:28 | ~41 sec | `linux` | [:package: zip](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-190726-052146-20bd0d.zip) |
| :heavy_check_mark: | 20bd0ddc | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1540/2/) | 2019-07-26 05:26:02 | ~4 min | `ios` | [:package: zip](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-190726-052145-20bd0d.zip) |
| :heavy_check_mark: | 20bd0ddc | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1540/2/) | 2019-07-26 05:27:26 | ~5 min | `android` | [:package: aar](https://status-go.ams3.digitaloceanspaces.com/status-go-android-190726-052144-20bd0d.aar) |
| | | | | | | |
| :heavy_check_mark: | a88e0902 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1540/3/) | 2019-08-06 08:46:31 | ~39 sec | `linux` | [:package: zip](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-190806-084551-a88e09.zip) |
| :heavy_check_mark: | a88e0902 | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1540/3/) | 2019-08-06 08:49:44 | ~3 min | `ios` | [:package: zip](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-190806-084550-a88e09.zip) |
| :heavy_check_mark: | a88e0902 | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1540/3/) | 2019-08-06 08:52:24 | ~6 min | `android` | [:package: aar](https://status-go.ams3.digitaloceanspaces.com/status-go-android-190806-084549-a88e09.aar) |
| :heavy_check_mark: | a88e0902 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1540/4/) | 2019-08-08 06:27:06 | ~51 sec | `linux` | [:package: zip](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-190808-062614-a88e09.zip) |
| :heavy_check_mark: | a88e0902 | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1540/4/) | 2019-08-08 06:29:24 | ~3 min | `ios` | [:package: zip](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-190808-062613-a88e09.zip) |
| :heavy_check_mark: | a88e0902 | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1540/4/) | 2019-08-08 06:31:39 | ~5 min | `android` | [:package: aar](https://status-go.ams3.digitaloceanspaces.com/status-go-android-190808-062612-a88e09.aar) |
| | | | | | | |
| :heavy_check_mark: | 7729a8f7 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1540/5/) | 2019-08-08 06:33:49 | ~39 sec | `linux` | [:package: zip](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-190808-063309-7729a8.zip) |
| :heavy_check_mark: | 7729a8f7 | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1540/5/) | 2019-08-08 06:36:31 | ~3 min | `ios` | [:package: zip](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-190808-063308-7729a8.zip) |
| :heavy_check_mark: | 7729a8f7 | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1540/5/) | 2019-08-08 06:38:27 | ~5 min | `android` | [:package: aar](https://status-go.ams3.digitaloceanspaces.com/status-go-android-190808-063307-7729a8.aar) |
| | | | | | | |
| :heavy_check_mark: | 9c90f0f2 | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1540/6/) | 2019-08-08 06:38:21 | ~38 sec | `linux` | [:package: zip](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-190808-063742-9c90f0.zip) |
| :heavy_check_mark: | 9c90f0f2 | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1540/6/) | 2019-08-08 06:40:19 | ~2 min | `ios` | [:package: zip](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-190808-063741-9c90f0.zip) |
| :heavy_check_mark: | 9c90f0f2 | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1540/6/) | 2019-08-08 06:43:55 | ~5 min | `android` | [:package: aar](https://status-go.ams3.digitaloceanspaces.com/status-go-android-190808-063829-9c90f0.aar) |
| | | | | | | |
| :heavy_check_mark: | eafa1d93 | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1540/7/) | 2019-08-08 06:44:16 | ~56 sec | `linux` | [:package: zip](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-190808-064319-eafa1d.zip) |
| :heavy_check_mark: | eafa1d93 | [#7](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1540/7/) | 2019-08-08 06:46:26 | ~3 min | `ios` | [:package: zip](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-190808-064318-eafa1d.zip) |
| :heavy_check_mark: | eafa1d93 | [#7](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1540/7/) | 2019-08-08 06:49:02 | ~5 min | `android` | [:package: aar](https://status-go.ams3.digitaloceanspaces.com/status-go-android-190808-064357-eafa1d.aar) |
| | | | | | | |
| :heavy_check_mark: | 8cc6353f | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1540/8/) | 2019-08-12 06:56:31 | ~44 sec | `linux` | [:package: zip](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-190812-065546-8cc635.zip) |
| :heavy_check_mark: | 8cc6353f | [#8](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1540/8/) | 2019-08-12 06:58:27 | ~2 min | `ios` | [:package: zip](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-190812-065545-8cc635.zip) |
| :heavy_check_mark: | 8cc6353f | [#8](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1540/8/) | 2019-08-12 07:03:11 | ~7 min | `android` | [:package: aar](https://status-go.ams3.digitaloceanspaces.com/status-go-android-190812-065544-8cc635.aar) |
| :x: | 8cc6353f | [#9](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1540/9/) | 2019-08-20 16:03:53 | ~16 sec | `ios` | [:page_facing_up: log](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1540/9/consoleText) |
| :x: | 8cc6353f | [#9](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1540/9/) | 2019-08-20 16:03:57 | ~22 sec | `android` | [:page_facing_up: log](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1540/9/consoleText) |
| :heavy_check_mark: | 8cc6353f | [#9](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1540/9/) | 2019-08-20 16:04:20 | ~43 sec | `linux` | [:package: zip](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-190820-160337-8cc635.zip) |
| | | | | | | |
| :x: | 1b32e3e3 | [#10](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1540/10/) | 2019-08-21 06:11:15 | ~25 sec | `ios` | [:page_facing_up: log](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1540/10/consoleText) |
| :x: | 1b32e3e3 | [#10](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1540/10/) | 2019-08-21 06:11:20 | ~31 sec | `android` | [:page_facing_up: log](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1540/10/consoleText) |
| :heavy_check_mark: | 1b32e3e3 | [#10](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1540/10/) | 2019-08-21 06:11:46 | ~56 sec | `linux` | [:package: zip](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-190821-061050-1b32e3.zip) |
| | | | | | | |
| :heavy_check_mark: | 350c909c | [#11](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1540/11/) | 2019-08-21 06:44:06 | ~55 sec | `linux` | [:package: zip](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-190821-064310-350c90.zip) |
| :heavy_check_mark: | 350c909c | [#11](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1540/11/) | 2019-08-21 06:49:29 | ~6 min | `android` | [:package: aar](https://status-go.ams3.digitaloceanspaces.com/status-go-android-190821-064308-350c90.aar) |
| :heavy_check_mark: | 350c909c | [#11](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1540/11/) | 2019-08-21 06:49:41 | ~6 min | `ios` | [:package: zip](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-190821-064309-350c90.zip) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 92c966e6 | [#12](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1540/12/) | 2019-08-21 06:44:48 | ~39 sec | `linux` | [:package: zip](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-190821-064408-92c966.zip) |
| :heavy_check_mark: | 92c966e6 | [#12](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1540/12/) | 2019-08-21 06:52:28 | ~2 min | `ios` | [:package: zip](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-190821-064942-92c966.zip) |
| :heavy_check_mark: | 92c966e6 | [#12](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1540/12/) | 2019-08-21 06:55:11 | ~5 min | `android` | [:package: aar](https://status-go.ams3.digitaloceanspaces.com/status-go-android-190821-064931-92c966.aar) |
| | | | | | | |
| :heavy_check_mark: | ebf55a6b | [#13](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1540/13/) | 2019-08-21 07:30:42 | ~1 min | `linux` | [:package: zip](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-190821-072941-ebf55a.zip) |
| :heavy_check_mark: | ebf55a6b | [#13](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1540/13/) | 2019-08-21 07:32:39 | ~2 min | `ios` | [:package: zip](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-190821-072940-ebf55a.zip) |
| :heavy_check_mark: | ebf55a6b | [#13](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1540/13/) | 2019-08-21 07:35:31 | ~5 min | `android` | [:package: aar](https://status-go.ams3.digitaloceanspaces.com/status-go-android-190821-072939-ebf55a.aar) |

adambabik commented on services/wallet/database.go:
```diff
@@ -111,6 +96,10 @@ func (blob *JSONBlob) Value() (driver.Value, error) {
 }
 
+func NewDB(db *sql.DB) *Database {
+	return &Database{db}
```
With this PR we are vulnerable to the problem described in https://github.com/status-im/status-go/issues/1537?

dshulyak replied:
yes that part wasn't added yet

dshulyak replied:
@adambabik made a separate pr to resolve that problem (https://github.com/status-im/status-go/pull/1564)

dshulyak committed:
Wallet database isolated by the network id

Wallet database refactored so that every query ensures isolation by the network id.
Network id provided when database object is created, thus it is transparent to other parts
of the wallet module.

Additionally every uniqueness index is changed to ensure that it doesn't prevent adding
object with same id but from a different network.

yenda commented:
what is the status on this one? can it be merged?

dshulyak commented:
@yenda it is based on accounts-persistence branch. once we will merge that one - i will start merging all smaller changes
